syntax = "proto3";

import "google/protobuf/timestamp.proto";

// Criação de Leilão ------------------------
message CreateAuctionRequest {
    string description = 1;
    google.protobuf.Timestamp start_time = 2;
    google.protobuf.Timestamp end_time = 3;
}

message Auction {
    int32 id = 1;
    string description = 2;
    google.protobuf.Timestamp start_time = 3;
    google.protobuf.Timestamp end_time = 4;
    string status = 5;
}

// Listagem de Leilão -----------------------
message ListAuctionsRequest {}

message ListAuctionsResponse {
  repeated Auction auctions = 1;
}

enum AuctionStatus {
    AUCTION_STATUS_ACTIVE = 0;
    AUCTION_STATUS_PENDING = 1;
    AUCTION_STATUS_CLOSED = 2;
}

service AuctionService {
  rpc CreateAuction(Auction) returns (Auction);
  rpc ListAuctions(ListAuctionsRequest) returns (ListAuctionsResponse);
}